### PR TITLE
FAQ, sql: improve isolation level and FAQ wording

### PR DESCRIPTION
**Pull request by lilin90**
This PR tries to improve and correct some wording.

@shenli @siddontang @morgo @queenypingcap @dcalvin PTAL

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -44,11 +44,15 @@ TiDB is self-healing. All of the three components, TiDB, TiKV and PD, can tolera
 
 #### How is TiDB strongly consistent?
 
-TiDB uses the [Raft consensus algorithm](https://raft.github.io/) to ensure consistency among multiple replicas. At the bottom layer, TiDB uses a model of replication log + State Machine to replicate data. For the write requests, the data is written to a Leader and the Leader then replicates the command to its Followers in the form of log. When the majority of nodes in the cluster receive this log, this log is committed and can be applied into the State Machine. TiDB has the latest data even if a minority of the replicas are lost.
+TiDB implements the Repeatable Read isolation in the Snapshot Isolation level. It does so by using the [Raft consensus algorithm](https://raft.github.io/) to ensure consistency among multiple replicas.
```
Comment (morgo) [inline FAQ.md]: Perhaps something like:
```suggestion
TiDB implements Snapshot Isolation consistency, which it advertises as `REPEATABLE-READ` for compatibility with MySQL. Data is redundantly copied between TiKV nodes using the [Raft consensus algorithm](https://raft.github.io/) to ensure recoverability should a node failure occur.
```
The point I am making here is that raft doesn't provide consistency unless there is a failure event.  It is the transactional protocol on top of Raft that provides it.  Raft provides recoverability.

File: sql/transaction-isolation.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ The SQL-92 standard defines four levels of transaction isolation: Read Uncommitt
 | Repeatable Read  | Not possible | Not possible       | Not possible in  TiDB | Possible              |
 | Serializable     | Not possible | Not possible       | Not possible          | Not possible          |
 
-TiDB offers the Repeatable Read isolation level.
+TiDB implements the Repeatable Read isolation in the Snapshot Isolation level. However, the TiDB Repeatable Read isolation level differs from the [ANSI Repeatable Read isolation level](/sql/transaction-isolation.md#difference-between-tidb-and-ansi-repeatable-read) and the [MySQL Repeatable Read level](/sql/transaction-isolation.md#difference-between-tidb-and-mysql-repeatable-reads).
```
Comment (morgo) [inline sql/transaction-isolation.md]:
```suggestion
TiDB implements Snapshot Isolation consistency, which it advertises as `REPEATABLE-READ` for compatibility with MySQL. This differs from the [ANSI Repeatable Read isolation level](/sql/transaction-isolation.md#difference-between-tidb-and-ansi-repeatable-read) and the [MySQL Repeatable Read level](/sql/transaction-isolation.md#difference-between-tidb-and-mysql-repeatable-reads).
```

File: FAQ.md [hunk 3]
```diff
@@ -44,11 +44,15 @@ TiDB is self-healing. All of the three components, TiDB, TiKV and PD, can tolera
...
+
+At the bottom layer, TiDB uses a model of replication log + State Machine to replicate data. For the write requests, the data is written to a Leader and the Leader then replicates the command to its Followers in the form of log. When the majority of nodes in the cluster receive this log, this log is committed and can be applied into the State Machine.
```
Comment (morgo) [inline FAQ.md]:
```suggestion
At the bottom layer, TiKV uses a model of replication log + State Machine to replicate data. For the write requests, the data is written to a Leader and the Leader then replicates the command to its Followers in the form of log. When the majority of nodes in the cluster receive this log, this log is committed and can be applied into the State Machine.
```

**Comment by lilin90** [comment]
@shenli @siddontang PTAL

File: FAQ.md [hunk 4]
```diff
@@ -447,7 +451,7 @@ This is because the `--initial-cluster` in the PD startup parameter contains a m
 
 #### What's the maximum tolerance for time synchronization error of PD?
 
-Theoretically, the smaller of the tolerance, the better. During leader changes, if the clock goes back, the process won't proceed until it catches up with the previous leader. PD can tolerate any synchronization error, but a larger error value means a longer period of service stop during the leader change.
+Theoretically, the smaller of the time synchronization error, the better. PD can tolerate any synchronization error, but a larger error value means a longer period of service stop during the leader change. If the clock goes back during a leader change, the change process will not proceed until it catches up with the previous leader.
```
Comment (Hoverbear) [inline FAQ.md]:
```suggestion
Theoretically, the smaller the time synchronization error, the better. PD can tolerate any synchronization error, but a larger error value means a longer period of service stop during the leader change. If the clock goes back during a leader change, the change process will not proceed until it catches up with the previous leader.
```

**Comment by lilin90** [comment]
@Hoverbear Would you please sign the CLA? Thanks~

File: FAQ.md [hunk 5]
```diff
@@ -44,11 +44,15 @@ TiDB is self-healing. All of the three components, TiDB, TiKV and PD, can tolera
 
 #### How is TiDB strongly consistent?
 
-TiDB uses the [Raft consensus algorithm](https://raft.github.io/) to ensure consistency among multiple replicas. At the bottom layer, TiDB uses a model of replication log + State Machine to replicate data. For the write requests, the data is written to a Leader and the Leader then replicates the command to its Followers in the form of log. When the majority of nodes in the cluster receive this log, this log is committed and can be applied into the State Machine. TiDB has the latest data even if a minority of the replicas are lost.
+TiDB implements Snapshot Isolation consistency, which it advertises as `REPEATABLE-READ` for compatibility with MySQL. Data is redundantly copied between TiKV nodes using the [Raft consensus algorithm](https://raft.github.io/) to ensure recoverability should a node failure occur.
```
Comment (siddontang) [inline FAQ.md]: IMO, SI is a little different from RR.

Comment (morgo) [reply FAQ.md]: @siddontang it does not claim that SI is the same as RR.  It says that it _advertises_ it as RR for compatibility.

**Comment by lilin90** [comment]
@morgo @siddontang @Hoverbear Is this PR ready to be merged? Please leave your review feedback. Thanks!

**Pull request merged**